acra/acra-in-depth/architecture/dbs.md

```diff
@@ -4,31 +4,61 @@ bookCollapseSection: true
 weight: 4
 ---
 
-**TODO: Expand**
+# Key storage
 
-# Key storage 
-
-Is used to store encrypted keys and tokens. 
+Is used to store encrypted keys and tokens.
 
 ## How it works
 
+A dedicated database (we currently support Redis for this purpose) is brought up.
+AcraServer and/or AcraTranslator are configured to connect to it and use it for key/token storage.
+
+When AcraServer or AcraTranslator need a key for specific client ID or zone, they ask key storage for it.
```

Comment:
better to abstract from clientID/zoneID and reference to purposes. Acra has separate keys for each purposes: transport, symmetric encryption, asymmetric, logging, hashing, etc.